pkg/authn/verifier.go

```diff
@@ -85,6 +93,20 @@ func (v *RegistryVerifier) validateAudience(token *jwt.Token) error {
 	return fmt.Errorf("could not find node ID in audience %v", audience)
 }
 
+func (v *RegistryVerifier) validateClaims(token *jwt.Token) error {
+	claims, ok := token.Claims.(*XmtpdClaims)
+	if !ok {
+		return fmt.Errorf("invalid token claims type")
+	}
+
+	// Check if the token is valid
+	if !token.Valid {
+		return fmt.Errorf("invalid token")
+	}
```

Comment:
Should we move this check before the type assertion? Checking the token is valid should be a prerequisite before checking the claims.